packages/website/docs/sending-algolia-insights-events.md

```diff
@@ -0,0 +1,298 @@
+
+### Changing the `eventName`
+
+This snippet shows how to instantiate a plugin that sends a click event with "Product Selected from Autocomplete" as the `eventName` whenever a user selects an item. You may want to do this to differentiate between click events sent from your Autocomplete vs. other parts of your search implementation.
+
+```js title="index.js"
+const appId = "latency";
+const apiKey = "6be0576ff61c053d5f9a3225e2a90f76";
+const searchClient = algoliasearch(appId, apiKey);
+insightsClient("init", { appId, apiKey });
+
+const algoliaInsightsPlugin = createAlgoliaInsightsPlugin({
+  insightsClient,
+  onSelect({ insights, insightsEvents }) {
+    const events = insightsEvents.map((insightsEvent) => ({
+      ...insightsEvent,
+      eventName: 'Product Selected from Autocomplete',
+    }));
+    insights.clickedObjectIDsAfterSearch(...events);
+  },
+})
+```
+
+### Sending different events for different result types
+
+If you're [using multiple different Algolia indices in the same autocomplete](combining-multiple-result-types), for example one for products and one for suggestions, you may want to use different `eventName`s for each section. You can change this (and other parts of the event) conditionally based on the source index name:
+
+```js title="index.js"
+const appId = "latency";
+const apiKey = "6be0576ff61c053d5f9a3225e2a90f76";
+const searchClient = algoliasearch(appId, apiKey);
+insightsClient("init", { appId, apiKey });
+
+const algoliaInsightsPlugin = createAlgoliaInsightsPlugin({
+  insightsClient,
+  onSelect({ insights, insightsEvents }) {
+    const events = insightsEvents.map((insightsEvent) => {
+      switch (insightsEvent.index) {
+        case 'instant_search_demo_query_suggestions': {
+          return {
+            ...insightsEvent,
+            eventName: 'Suggestion Selected from Autocomplete',
+          };
+        }
+        case 'instant_search': {
+          return {
+            ...insightsEvent,
+            eventName: 'Product Selected from Autocomplete',
+          };
+        }
+        default: {
+          return {
+            ...insightsEvent,
+            eventName: 'Item Selected from Autocomplete',
+          };
+        }
+      }
+    });
+    insights.clickedObjectIDsAfterSearch(...events);
+  },
+});
+```
+
+## Sending events from templates
+
+Though the default Insights plugin doesn't send any conversion events, you may want to. For example, you may have created a [template](templates) with an "Add to cart" button.
+
+![Autocomplete with add to cart button](/img/add-to-cart.jpg)
+
+If a user adds an item to their shopping cart directly from the autocomplete, you can send a conversion event directly from your template:
+
+```jsx title="ProductItem.jsx"
+function ProductItem({ hit, insights }) {
```

Comment:
We need to show users how to get the `insights` value, which is from the context in the templates:

```ts
autocomplete({
  // ...
  plugins: [algoliaInsightsPlugin],
  templates: {
    item({ item, state }) {
      return (
        <ProductItem
          hit={item}
          insights={state.context.algoliaInsightsPlugin.insights}
        />
      );
    },
  },
});
```

Reply:
Good point! Added here: https://github.com/algolia/autocomplete.js/pull/430/commits/348b8c7d316e08aecc836786e45be47c89af1139